domain/src/main/kotlin/no/nav/su/se/bakover/domain/vedtak/Vedtak.kt

```diff
@@ -0,0 +1,132 @@
+package no.nav.su.se.bakover.domain.vedtak
+
+import no.nav.su.se.bakover.common.Tidspunkt
+import no.nav.su.se.bakover.common.UUID30
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.domain.Fnr
+import no.nav.su.se.bakover.domain.NavIdentBruker
+import no.nav.su.se.bakover.domain.Saksnummer
+import no.nav.su.se.bakover.domain.behandling.Behandlingsinformasjon
+import no.nav.su.se.bakover.domain.beregning.Beregning
+import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.EksterneIverksettingsstegEtterUtbetaling
+import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.EksterneIverksettingsstegForAvslag
+import no.nav.su.se.bakover.domain.oppdrag.simulering.Simulering
+import no.nav.su.se.bakover.domain.revurdering.IverksattRevurdering
+import no.nav.su.se.bakover.domain.søknadsbehandling.Søknadsbehandling
+import java.util.UUID
+
+interface IBehandling {
+    val id: UUID
+    val sakId: UUID
+    val saksnummer: Saksnummer
+    val fnr: Fnr
+}
+
+interface IVedtak {
+    val id: UUID
+    val opprettet: Tidspunkt
+    val behandling: IBehandling
+    val behandlingsinformasjon: Behandlingsinformasjon
+}
+
+interface IVedtakSomGirUtbetaling : IVedtak {
```

Comment:
Ser ikke helt poenget med IVedtakSomGirUtbetaling og IVedtakSomIkkeGirUtbetaling i tillegg til subtypene av sealed classen. Er dette forarbeid for andre behandlingstyper som kommer utenfor sealed class Vedtak? I så fall synes jeg vi skal kalle sealed class Vedtak for noe mer spesifikt. Og kalle IVedtak for Vedtak. Hvis ikke foreslår jeg at de slettes.

Reply:
Noen av disse interfacene ble egentlig litt sånn fordi vi drev og testet noen greier med delegation osv. Vi kan ta en ny titt på det og evt. stramme opp litt.